homeassistant/components/tesla_fleet/climate.py

```diff
@@ -0,0 +1,330 @@
+"""Climate platform for Tesla Fleet integration."""
+
+from __future__ import annotations
+
+from itertools import chain
+from typing import Any, cast
+
+from tesla_fleet_api.const import CabinOverheatProtectionTemp, Scope
+
+from homeassistant.components.climate import (
+    ATTR_HVAC_MODE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_HALVES,
+    PRECISION_WHOLE,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TeslaFleetConfigEntry
+from .const import DOMAIN, TeslaFleetClimateSide
+from .entity import TeslaFleetVehicleEntity
+from .helpers import handle_vehicle_command
+from .models import TeslaFleetVehicleData
+
+DEFAULT_MIN_TEMP = 15
+DEFAULT_MAX_TEMP = 28
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: TeslaFleetConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Tesla Fleet Climate platform from a config entry."""
+
+    async_add_entities(
+        chain(
+            (
+                TeslaFleetClimateEntity(
+                    vehicle, TeslaFleetClimateSide.DRIVER, entry.runtime_data.scopes
+                )
+                for vehicle in entry.runtime_data.vehicles
+            ),
+            (
+                TeslaFleetCabinOverheatProtectionEntity(
+                    vehicle, entry.runtime_data.scopes
+                )
+                for vehicle in entry.runtime_data.vehicles
+            ),
+        )
+    )
+
+
+class TeslaFleetClimateEntity(TeslaFleetVehicleEntity, ClimateEntity):
+    """Tesla Fleet vehicle climate entity."""
+
+    _attr_precision = PRECISION_HALVES
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF]
+    _attr_supported_features = (
+        ClimateEntityFeature.TURN_ON
+        | ClimateEntityFeature.TURN_OFF
+        | ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_preset_modes = ["off", "keep", "dog", "camp"]
+    _enable_turn_on_off_backwards_compatibility = False
+
+    def __init__(
+        self,
+        data: TeslaFleetVehicleData,
+        side: TeslaFleetClimateSide,
+        scopes: Scope,
+    ) -> None:
+        """Initialize the climate."""
+
+        self.read_only = Scope.VEHICLE_CMDS not in scopes or data.signing
```

Comment:
Both the scopes and signing information are only updated in init. They change so infrequently in the real world that I think its fair to ask the user to reload the integration rather than have it automatically poll / look for changes at any sort of interval.

Reply:
I guess you're right. I don't like the need of manually needing to reload or things if not absolutely necessary but on this occasion I might agree with you.